dev-packages/browser-integration-tests/suites/tracing/dsc-txn-name-update/test.ts

```diff
@@ -0,0 +1,164 @@
+import { expect } from '@playwright/test';
+import type { Page } from '@playwright/test';
+import type { DynamicSamplingContext } from '@sentry/types';
+import { sentryTest } from '../../../utils/fixtures';
+import type { EventAndTraceHeader } from '../../../utils/helpers';
+import {
+  eventAndTraceHeaderRequestParser,
+  getMultipleSentryEnvelopeRequests,
+  shouldSkipTracingTest,
+} from '../../../utils/helpers';
+
+sentryTest('updates the DSC when the txn name is updated and high-quality', async ({ getLocalTestUrl, page }) => {
+  if (shouldSkipTracingTest()) {
+    sentryTest.skip();
+  }
+
+  const url = await getLocalTestUrl({ testDir: __dirname });
+
+  await page.goto(url);
+
+  /*
+  Test Steps:
+  1. Start new span with LQ txn name (source: url)
+  2. Make request and check that baggage has no transaction name
+  3. Capture error and check that envelope trace header has no transaction name
+  4. Update span name and source to HQ (source: route)
+  5. Make request and check that baggage has HQ txn name
+  6. Capture error and check that envelope trace header has HQ txn name
+  7. Update span name again with HQ name (source: route)
+  8. Make request and check that baggage has updated HQ txn name
+  9. Capture error and check that envelope trace header has updated HQ txn name
+  10. End span and check that envelope trace header has updated HQ txn name
```

Comment:
can we add 11. to send a request after the span has ended, and make sure what is sent then? as we keep the trace alive after that (but not on the active span anymore, but the propagation context) this is also a scenario, right?

Reply:
yup, good point. Let me add that

Reply:
Hmm so it looks like at the moment, the DSC is also re-generated and no span information is added at all, meaning we not only loose `transaction` name but also `sample_rate` and `sampled`

Reply:
argh this is also problematic for the `sentry-trace` header. In the same scenario, we omit the sampling decision:

![image](https://github.com/user-attachments/assets/e5640c44-bebd-4651-9f4d-39edec9d2264)

(first 3 requests are happening while the span is active, the last one afterwards. Same traceId but different sampled flags)

Reply:
let's reflect this in this PR I'd say and then see if we can fix/improve this in a follow up...?

Reply:
@mydea I think the behavior in this test is caused by the constructed scenario of us starting the trace by creating a custom span as opposed to a pageload or navigation span. I will look into what exactly is causing the transaction info to be missing while the traceId remains the same in this scenario. However, I added tests for the usual pageload transaction scenario where after the transaction ends we make requests/send additional spans. They have consistent DSC data from the initial pageload transaction.